src/cascadia/TerminalApp/Profile.cpp

```diff
@@ -49,7 +49,13 @@ static const std::wstring CURSORSHAPE_FILLEDBOX{ L"filledBox" };
 static const std::wstring CURSORSHAPE_EMPTYBOX{ L"emptyBox" };
 
 Profile::Profile() :
-    _guid{},
+    Profile(winrt::guid{})
+{
+    UuidCreate(&_guid);

```

Comment:
`UuidCreate` should probably be replaced with something like `CoCreateGuid`. It should also probably check/log the result code somehow.

Reply:
I'm specifically citing `https://stackoverflow.com/questions/12354737/how-do-you-create-a-new-guid-in-c-in-winrt` which says that `CoCreateGuid` is supported in our universal SDK and `UuidCreate` is a desktop only API, which leads me to believe `CoCreateGuid` is supported in an APIset.

The WCD confirms this:
- `api-ms-win-core-com-l1!CoCreateGuid` holds `CoCreateGuid`

whereas `UuidCreate` is only available in things without APIset like:
- `rpcrt4.dll!UuidCreate`

Reply:
Agreed; I discovered this in parallel while working on #897.

Reply:
per the comment below, I think @DHowett-MSFT should really add a types library function that is of signature
```
GUID GimmieDatGuid();
```
that's throwable and route all guid-making calls through that.